chainstate/src/detail/chainstateref.rs

```diff
@@ -547,10 +551,12 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
     }
 
     fn make_cache_with_disconnected_transactions(
-        &self,
+        &'a self,
+        utxo_view: &'a impl UtxosView,
```

Comment:
Because a ref to the utxo_view is stored in tx_verifier which is returned from this function by value. So utxo_view should live longer than that.

Reply:
In that case, how about we put the UtxosDB inside of transaction verifier, since it's just a thin wrapper around db_tx, which is already in the transaction verifier?

Reply:
The problem with that is that I'm not sure it's possible in rust to initialize struct fields where one is a reference to another. And in this case utxo_cache has a reference to utxo_db

Reply:
As long as the lifetime is proper and you're borrowing only immutable references, there won't be an issue.

Reply:
I mean I literally don't know how to do that syntactically:
```rust
impl<'a, S: UtxosStorageRead> TransactionVerifier<'a, S> {
    pub fn new(db_tx: &'a S, chain_config: &'a ChainConfig) -> Self {
        Self {
            db_tx,
            chain_config,
            tx_index_cache: BTreeMap::new(),
            utxo_db: UtxosDB::new(db_tx),
            utxo_cache: utxo_db???.derive_cache(), //self is not ready and utxo_db field is not accessable
            utxo_block_undo: BTreeMap::new(),
        }
    }
}
```

something like this doesn't work either:
```rust
impl<'a, S: UtxosStorageRead> TransactionVerifier<'a, S> {
    pub fn new(db_tx: &'a S, chain_config: &'a ChainConfig) -> Self {
        let utxo_db = UtxosDB::new(db_tx);
        let utxo_cache = utxo_db.derive_cache(); //borrows var that is moved
        Self {
            db_tx,
            chain_config,
            tx_index_cache: BTreeMap::new(),
            utxo_db,
            utxo_cache,
            utxo_block_undo: BTreeMap::new(),
        }
    }
}
```

Reply:
discussed, leaving a TODO to investigate later